### PR TITLE
template improvements

### PR DESCRIPTION
**Pull request by otakup0pe**
Handles #25 Needs more tests/docs before ready to merge. Related to this is a bunch of command line options around the rendering of secret key names.

This PR has a bunch of changes around testing because the travis-ci builds keep failing in new and exciting ways. Also I added a pile of (mostly integration) tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 all: test package
 
-package:
+package: version
+	cp version aomi/version
```
Comment (ericfode) [inline Makefile]: 👏 Yay makefiles

File: README.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ The `aomi` tool has several requirements which can (generally) all be sourced fr
 
 The [PyYAML](http://pyyaml.org/) package, by default, will make use of libyaml. This can be a problem on some systems as you may need to manually install libyaml.
 
+You should be using a recent enough version of Python to have good TLS support. Vault can make use of SNI and that requires fresh Python 2.7.
```
Comment (ericfode) [inline README.md]: Python 3?

File: README.md [hunk 3]
```diff
@@ -170,41 +188,56 @@ This action takes two arguments - the source path and the destination file. The
...
+
+## environment
+
+This action takes any number of Vault paths are it's arguments. In return, it will generate a small snipped exporting the contained secrets as environment variables. This output is sufficient to be eval'd (no really, don't do this) or piped to a file an sourced in to a shell. Export snippets can be generated  with `--export`.
 
 ```
-aomi environment foo/bar/baz --prefix foo/bar
-BAZ_USER="foo"
-BAZ_PASSWORD="bar"
+aomi environment foo/bar/baz
```
Comment (ericfode) [inline README.md]: Does prefix work anymore?

File: README.md [hunk 4]
```diff
@@ -170,41 +188,56 @@ This action takes two arguments - the source path and the destination file. The
...
+FOO_BAR_BAZ_USER="foo"
+FOO_BAR_BAZ_PASSWORD="bar"
 ```
 
 ## template
```
Comment (ericfode) [inline README.md]: What about a usage example of templating?

File: aomi/seed.py [hunk 5]
```diff
@@ -29,22 +30,35 @@ def ensure_mounted(client, backend, mount):
...
 def var_file(client, secret, opt):
     """Seed a var_file into Vault"""
     path = "%s/%s" % (secret['mount'], secret['path'])
     var_file_name = hard_path(secret['var_file'], opt.secrets)
+    secret_file(var_file_name)
     varz = yaml.load(open(var_file_name).read())
     if 'var_file' not in secret \
        or 'mount' not in secret \
        or 'path' not in secret:
         problems("Invalid generic secret definition %s" % secret)
```
Comment (ericfode) [inline aomi/seed.py]: Maybe say what is missing in the problems. Maybe break this validate if statement in to a function that returns an error message if there is any problems.

File: aomi/seed.py [hunk 6]
```diff
@@ -90,13 +107,17 @@ def aws(client, secret, opt):
     region = aws_region(secret, aws_obj)
 
     aws_path = "%s/config/root" % secret['mount']
-    if not is_tagged(secret.get('tags', []), opt.tags):
+    if not is_tagged(opt.tags, secret.get('tags', [])):
         log("Skipping %s as it does not have appropriate tags" %
```
Comment (ericfode) [inline aomi/seed.py]: What tags are appropriate?

File: aomi/seed.py [hunk 7]
```diff
@@ -185,24 +207,49 @@ def app(client, app_obj, opt):
     else:
         name = os.path.splitext(os.path.basename(app_obj['app_file']))[0]
 
-    if not is_tagged(app_obj.get('tags', []), opt.tags):
+    if not is_tagged(opt.tags, app_obj.get('tags', [])):
         log("Skipping %s as it does not have appropriate tags" % name, opt)
```
Comment (ericfode) [inline aomi/seed.py]: What tags are appropriate?

File: aomi/seed.py [hunk 8]
```diff
@@ -185,24 +207,49 @@ def app(client, app_obj, opt):
...
-    if 'app_id' not in data \
-       or ('policy' not in data and 'policy_name' not in data):
+
+    ensure_auth(client, 'app-id')
+    if opt.mount_only:
+        log("Only enabling app-id", opt)
+        return
+
+    if 'users' not in data:
         problems("Invalid app file %s" % app_file)
```
Comment (ericfode) [inline aomi/seed.py]: Why is the file invalid. Maybe follow the pattern of 

Input -> Validator -> Maybe Problems 

again.

File: aomi/seed.py [hunk 9]
```diff
@@ -232,25 +279,29 @@ def files(client, secret, opt):
 
     obj = {}
     vault_path = "%s/%s" % (secret['mount'], secret['path'])
-    if not is_tagged(secret.get('tags', []), opt.tags):
+    if not is_tagged(opt.tags, secret.get('tags', [])):
         log("Skipping %s as it does not have appropriate tags" %
```
Comment (ericfode) [inline aomi/seed.py]: What tags were missing?

File: aomi/seed.py [hunk 10]
```diff
@@ -260,11 +311,40 @@ def policy(client, secret, opt):
...
     log('writing policy %s' % policy_name, opt)
     client.set_policy(policy_name, policy_data)
+
+
+def users(client, user_obj, opt):
+    """Creates userpass users in Vault"""
+    if 'password_file' not in user_obj or \
+       'username' not in user_obj or \
+       'policies' not in user_obj:
+        problems("Invalid user specification %s" % user_obj)
```
Comment (ericfode) [inline aomi/seed.py]: What is missing?

File: tests/integration/environment.bats [hunk 11]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bats
```
Comment (ericfode) [inline tests/integration/environment.bats]: 👏 This is awesome

File: README.md (hunk 2 above)
Comment (otakup0pe) [inline README.md]: Not in this PR https://github.com/Autodesk/aomi/issues/42

**Comment by otakup0pe** [comment]
Incorporated @ericfode 's feedback so will merge now... 🍰

**Pull request merged**